interactive/submit.py

```diff
@@ -50,18 +53,70 @@ def submit_analysis(
         force=force,
     )
 
-    job_request = project.interactive_workspace.job_requests.create(
+    create_job_request(analysis_request, backend, sha, project_yaml)
+
+    analysis_request.save()
+    notify_analysis_request_submitted(analysis_request)
+
+    return analysis_request
+
+
+def create_job_request(analysis_request, backend, sha, project_yaml):
+    workspace = analysis_request.project.interactive_workspace
+    job_request = workspace.job_requests.create(
         backend=backend,
-        created_by=creator,
+        created_by=analysis_request.created_by,
         sha=sha,
         project_definition=project_yaml,
         force_run_dependencies=True,
         requested_actions=["run_all"],
     )
-
     analysis_request.job_request = job_request
     analysis_request.save()
```

Comment:
Can you add a comment above this block to explain why we're updating the analysis_request here instead of at the call sites?

Reply:
So we don't forget?

Would you prefer it to be the callers responsibility?

Reply:
I've made it the callers responsibility